### DISCUSSION TITLE
Does Helix support CSHTML?

### DISCUSSION THREAD

**Discussion by kanielrkirby** [helix-editor/helix#11084] [General]
I occasionally for work need to code in a large legacy .NET project (but simply can't stand Visual Studio itself for editing). I wanted to edit some of the `.cshtml` files that are in it, but noticed there wasn't any syntax highlighting, and I haven't been able to find any reference to it, well, anywhere. I just want to take the opportunity to ask, is this something that's supported? Known about? Is there a workaround?

Thanks, everything else about Helix has been great, love what y'all have done so far.

**Comment by kirawi** [comment]
There is on tree-sitter grammar for it that's actively maintained. The only one, https://github.com/tree-sitter/tree-sitter-razor, last received a commit 8 years ago.

Your best bet would probably be adding `.cshtml` to the `html` file-types to get some form of highlighting.

**Reply by kanielrkirby** [reply]
Ahh, I see. Appreciate you checking that, probably should have looked there myself. It makes sense that there isn't a whole lot of crossover between the people using Treesitter, and the people doing work in .NET. Still surprising there's no updates in an entire 8 years. Anyways, thanks for the help!

**Comment by Benexl** [comment]
been sometime but incase anyone is still looking i found this and it works: https://github.com/slein-nl/tree-sitter-razor.git
